src/permutat.cc

```diff
@@ -1098,6 +1098,72 @@ static Obj FuncPermList(Obj self, Obj list)
     }
 }
 
+template <typename T>
+static inline Obj ListPerm_(Obj perm, UInt len)
+{
+    Obj                 res;            // handle of the image, result
+    Obj *               ptRes;          // pointer to the result
+    const T *           ptPrm;          // pointer to the permutation
+    UInt                lmp;            // largest moved point
+    UInt                i;              // loop variable
+
+    lmp = DEG_PERM<T>(perm);
```

Comment:
I guess from the variable name `lmp` that this is meant to be the largest moved point, right? If so, then this isn't correct I don't think. The degree of a permutation is the number of entries in the array storing the permutation, and this can be completely different from the largest moved point. For example if you do:

~~~
gap> x := (1, 70000);
(1,70000)
gap> y := x * x ^-1;
()
~~~
Then the degree of `y` is `70000` while the largest moved point is `0` (this certainly used to be the case, and the fact that `y` is in `IsPerm4Rep` seems to indicate that it still is).

Reply:
You are absolutely right. I'll add the test case and fix the PR. Thanks for carefully checking, much appreciated!

Reply:
fixed, see line 1112

Reply:
This fix isn't complete -- lmp can still be larger than len if len==0, and cause a segfault. For example:
```
x:=(1,2,3,10000);
y:=(1,10000);
ListPerm(x*y);
```
(This won't crash immediately, but will if you keep using GAP, as we have written off into as yet unused space).

Maybe just remove the `else` from `1113`, to make sure `lmp` is never larger than `len`?

Reply:
thanks, fixed now as well

Reply:
I've also revised the code some more, renamed `lmp`, etc.